packages/js/product-editor/src/blocks/section/edit.tsx

```diff
@@ -15,28 +17,91 @@ import {
  */
 import { sanitizeHTML } from '../../utils/sanitize-html';
 import { SectionBlockAttributes } from './types';
+import { hasAttributesUsedForVariations } from '../../utils';
 
 export function Edit( {
 	attributes,
-}: BlockEditProps< SectionBlockAttributes > ) {
+	context,
+}: BlockEditProps< SectionBlockAttributes > & {
+	context?: {
+		selectedTab?: string | null;
+	};
+} ) {
 	const { description, title, blockGap } = attributes;
+	const tab = context?.selectedTab || '';
 	const blockProps = useBlockProps();
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+	const isDisabledSection =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		[ 'inventory', 'pricing', 'shipping' ].includes( tab );
 	const innerBlockProps = useInnerBlocksProps(
 		{
 			className: classNames(
 				'wp-block-woocommerce-product-section__content',
-				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`
+				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`,
+				{
+					'is-disabled-section': isDisabledSection,
+				}
 			),
 		},
 		{ templateLock: 'all' }
 	);
 	const SectionTagName = title ? 'fieldset' : 'div';
 	const HeadingTagName = SectionTagName === 'fieldset' ? 'legend' : 'div';
 
+	useEffect( () => {
+		const elementsToReEnable: (
+			| HTMLInputElement
+			| HTMLButtonElement
+			| HTMLSelectElement
+			| HTMLTextAreaElement
+		 )[] = [];
+
+		if ( isDisabledSection ) {
+			const disabledSections = document.querySelectorAll(
+				'.is-disabled-section'
+			);
```

Comment:
This is not scoped to the nodes under the current section, so this will find all `.is-disabled-section` in the entire DOM. Some of which might not even be "our" sections (any extension could also use that general classname for something completely different).

Reply:
We don't need to scope the nodes, because the code won't run if `isDisabledSection` [is false](https://github.com/woocommerce/woocommerce/pull/40342/files#diff-1ef779b6cbb533c937b756d661e441c900f2a1523fde4b3eadec1a0d52553aa0R64).